daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md

```diff
@@ -63,103 +170,173 @@ spec:
-app = Flask(__name__)
+//code
 
-@app.route('/add', methods=['POST'])
-def add():
-    return "Added!"
+CancellationTokenSource source = new CancellationTokenSource();
+CancellationToken cancellationToken = source.Token;
+using var client = new DaprClientBuilder().Build();
+var result = client.CreateInvokeMethodRequest(HttpMethod.Get, "checkout", "checkout/" + orderId, cancellationToken);
+await client.InvokeMethodAsync(result);
 
-if __name__ == '__main__':
-    app.run()
 ```
+{{% /codetab %}}
 
-This Python app exposes an `add()` method via the `/add` endpoint.
 
-## Step 3: Invoke the service
+{{% codetab %}}
+```bash
 
-Dapr uses a sidecar, decentralized architecture. To invoke an application using Dapr, you can use the `invoke` API on any Dapr instance.
+//headers
 
-The sidecar programming model encourages each applications to talk to its own instance of Dapr. The Dapr instances discover and communicate with one another.
+import io.dapr.client.DaprClient;
+import io.dapr.client.DaprClientBuilder;
+import io.dapr.client.domain.HttpExtension;
+
+//code
 
-{{< tabs curl CLI >}}
+DaprClient daprClient = new DaprClientBuilder().build();
+var result = daprClient.invokeMethod(
+    "checkout",
+    "checkout/" + orderId,
+    null,
+    HttpExtension.GET,
+    String.class
+);
+
+```
+{{% /codetab %}}
 
 {{% codetab %}}
-From a terminal or command prompt run:
 ```bash
-curl http://localhost:3500/v1.0/invoke/cart/method/add -X POST
+
+//headers
+
+from dapr.clients import DaprClient
+
+//code
+
+with DaprClient() as daprClient:
+  result = daprClient.invoke_method(
+      "checkout",
+          f"checkout/{orderId}",
+          data=b'',
+          http_verb="GET"
+  )    
+
 ```
+{{% /codetab %}}
 
-Since the add endpoint is a 'POST' method, we used `-X POST` in the curl command.
+{{% codetab %}}
+```bash
 
-To invoke a 'GET' endpoint:
+//headers
+import (
+	dapr "github.com/dapr/go-sdk/client"
+)
 
+//code
+
+client, err := dapr.NewClient()
+if err != nil {
+  panic(err)
+}
+defer client.Close()
+ctx := context.Background()
+
+result, err := client.InvokeMethod(ctx, "checkout", "checkout/" + strconv.Itoa(orderId), "get") 
+
+```
+{{% /codetab %}}
+
+{{% codetab %}}
 ```bash
-curl http://localhost:3500/v1.0/invoke/cart/method/add
+
+//headers
+
+import { DaprClient, HttpMethod, CommunicationProtocolEnum } from 'dapr-client'; 
+
+//code
+
+const daprHost = "127.0.0.1"; 
+const client = new DaprClient(daprHost, process.env.DAPR_HTTP_PORT, CommunicationProtocolEnum.HTTP);
+const result = await client.invoker.invoke('checkout' , "checkout/" + orderId , HttpMethod.GET);   
+
 ```
+{{% /codetab %}}
 
-To invoke a 'DELETE' endpoint:
+{{% codetab %}}
+To invoke a 'GET' endpoint:
 
 ```bash
-curl http://localhost:3500/v1.0/invoke/cart/method/add -X DELETE
+curl http://localhost:3602/v1.0/invoke/checkout/method/checkout/100
 ```
 
-Dapr puts any payload returned by the called service in the HTTP response's body.
-
 ### Additional URL formats
```

Comment:
We should not include HTTP proxy in the curl example. That section should stand by itself as an option for customers looking to adopt Dapr coming from an application that already has http client code calling existing endpoints and they do not want to change over to using the invoke url format.

My recommendation is to leave the "Additional URL formats" section outside of the code blocks. Moving forward, we should elaborate on this feature